lib/validation.js

```diff
@@ -58,7 +58,31 @@ function validateOverride(configOverride) {
   }
 }
 
+function validateERLParams(params) {
+  const erlIsActiveKey = params.erlIsActiveKey; // redis' way of knowing whether erl is active or not
+  if (!erlIsActiveKey)  {
+    return new LimitdRedisValidationError('erlIsActiveKey is required for elevated limits', { code: 107 });
+  }
+}
+
+function validateConfigIsForElevatedBucket(key, bucketKeyConfig) {
+  if (!isConfigForElevatedBucket(bucketKeyConfig)) {
+    return new LimitdRedisValidationError(`Attempted to takeElevated() for a bucket with no elevated config. bucket:${key}, bucketKeyConfig:${bucketKeyConfig}`,
+      { code: 108 });
+  }
+}
+
+function isConfigForElevatedBucket(bucketKeyConfig) {
+  return bucketKeyConfig.elevated_limits
+    && bucketKeyConfig.elevated_limits.size
+    && bucketKeyConfig.elevated_limits.per_interval
+    && bucketKeyConfig.elevated_limits.erl_activation_period_seconds;
```

Comment:
Should this be validated, given that we have a default value assigned to it when not provided? (`ERL_DEFAULT_ACTIVATION_PERIOD_SECONDS`)

Maybe we should drop the default value then. (If you do so, remember to adjust the test buckets that don't have it configured at the moment)

Reply:
You're right it's provided by default, but I would rather keep that check as it would make tests fail if we accidentally remove the default value.

Reply:
Imo it's a good thing if the tests fail when we accidentally remove the default value. 🤔 We should either make it optional with a fallback default, or mandatory.

Reply:
I think we're saying the same thing. I want it to fail if the default is accidentally removed. This validation will make it fail.